docker-bake.hcl

```diff
@@ -39,6 +39,14 @@ target "lint" {
   }
 }
 
+target "generate" {
```

Comment:
we should have another target validating generation similar to buildkit one:
* https://github.com/moby/buildkit/blob/e15601a00fbef2805db1ed87be7bb88628ae926b/hack/dockerfiles/generated-files.Dockerfile#L42-L56
* https://github.com/moby/buildkit/blob/e15601a00fbef2805db1ed87be7bb88628ae926b/docker-bake.hcl#L172-L177

and add it to the validate job matrix: https://github.com/tonistiigi/fsutil/blob/43b9329361d9991d3b945ca13a6ace5f793b134a/.github/workflows/ci.yml#L23-L27

Reply:
Added this. I also redid part of the generate script because running protoc is very frustrating. We want the import path to be the fully qualified path so that buildkit can import `wire.proto` using the fully qualified path, but protoc was just not happy with the paths being different between projects so I just decided "we're doing this in a dockerfile anyway so I'll just write the protoc command there rather than go generate."